config/initializers/warden.rb

```diff
@@ -6,19 +6,24 @@
 require 'open_project/authentication/strategies/warden/user_basic_auth'
 require 'open_project/authentication/strategies/warden/session'
 
-strategies = {
-  basic_auth_failure: OpenProject::Authentication::Strategies::Warden::BasicAuthFailure,
-  global_basic_auth:  OpenProject::Authentication::Strategies::Warden::GlobalBasicAuth,
-  user_basic_auth:    OpenProject::Authentication::Strategies::Warden::UserBasicAuth,
-  session:            OpenProject::Authentication::Strategies::Warden::Session
-}
+WS = OpenProject::Authentication::Strategies::Warden
+
+strategies = [
+  [:basic_auth_failure, WS::BasicAuthFailure, 'Basic'],
+  [:global_basic_auth,  WS::GlobalBasicAuth,  'Basic'],
+  [:user_basic_auth,    WS::UserBasicAuth,    'Basic'],
```

Comment:
Yes, @NobodysNightmare's point is that we originally wanted to change it in order for basic auth login prompts not to pop up. We decided though that the APIv3 just as v2 should pay attention to the `X-Authentication-Scheme` *request* header and return the requested challenge if possible.

I.e. if requests to APIv3 use that header everything is fine. The question is whether it does. Since I can't even find APIv3 requests (everything I see seems to be /api/experimental) I'm not sure about that, though.